src/mesh/mesh_generation.C

```diff
@@ -956,9 +1014,25 @@ void MeshTools::Generation::build_cube(UnstructuredMesh & mesh,
               for (unsigned int k=0; k<=(2*nz); k++)
                 for (unsigned int j=0; j<=(2*ny); j++)
                   for (unsigned int i=0; i<=(2*nx); i++)
-                    mesh.add_point(Point(static_cast<Real>(i)/static_cast<Real>(2*nx),
-                                         static_cast<Real>(j)/static_cast<Real>(2*ny),
-                                         static_cast<Real>(k)/static_cast<Real>(2*nz)), node_id++);
+                  {
+                    const Node * const node =
+                        mesh.add_point(Point(static_cast<Real>(i) / static_cast<Real>(2 * nx),
+                                             static_cast<Real>(j) / static_cast<Real>(2 * ny),
+                                             static_cast<Real>(k) / static_cast<Real>(2 * nz)),
+                                       node_id++);
+                    if (k == 0)
```

Comment:
I guess it's done now, but any reason to not just call `BoundaryInfo::build_node_list_from_side_list()` some place at the end of `build_cube()`?

Reply:
I talked about this with @roystgnr in the beginning of working on this PR and he voiced his distaste for that method because it mixes two concepts that aren't necessarily related.

Reply:
Sorry I was unclear; I just dislike using that method to modify an arbitrary mesh; it would have made decent sense to use it in this case, to save you some work.

Although I do like the way you've done it here a tiny bit better; I've always imagined that some day we might modify some of our mesh reading and mesh generation capabilities so that, instead of clearing (or doing random garbage...) existing mesh data, they could merge new data in-place into an existing mesh.  Avoiding the shortcut here means that if we ever decide to do that then there's one less thing that will need to be fixed to make it work.